src/input/skeletal.rs

```diff
@@ -247,11 +228,133 @@ impl<C: openxr_data::Compositor> Input<C> {
             TransformedIt::Model(it) => convert(it, transforms),
         }
 
-        // TODO: This is an arbitrary transform, and only appears to work as expected in parent space.
-        transforms[Root as usize].position = Vec3::new(0.0, 0.0, -0.15).into();
-
         *self.skeletal_tracking_level.write().unwrap() = vr::EVRSkeletalTrackingLevel::Estimated;
     }
+
+    fn get_finger_state(&self, session_data: &SessionData, hand: Hand) -> FingerState {
+        let data = self.openxr.session_data.get();
+        let actions = &data
+            .input_data
+            .estimated_skeleton_actions
+            .get()
+            .unwrap()
+            .actions;
+        let subaction = match hand {
+            Hand::Left => self.openxr.left_hand.subaction_path,
+            Hand::Right => self.openxr.right_hand.subaction_path,
+        };
+        let input = SkeletalInputActionStates {
+            thumb_touch: actions
+                .thumb_touch
+                .state(&session_data.session, subaction)
+                .unwrap()
+                .current_state,
+            index_touch: actions
+                .index_touch
+                .state(&session_data.session, subaction)
+                .unwrap()
+                .current_state,
+            index_curl: actions
+                .index_curl
+                .state(&session_data.session, subaction)
+                .unwrap()
+                .current_state,
+            rest_curl: actions
+                .rest_curl
+                .state(&session_data.session, subaction)
+                .unwrap()
+                .current_state,
+        };
+
+        let index = input
+            .index_curl
+            .max(if input.index_touch || input.index_curl > 0.0 {
+                0.3
+            } else {
+                0.0
+            });
+
+        let mut state = self.estimated_finger_state[match hand {
+            Hand::Left => 0,
+            Hand::Right => 1,
+        }]
+        .lock()
+        .unwrap();
+
+        let current_time = std::time::Instant::now();
+
+        let target = FingerState {
+            index,
+            // Make other fingers curl with the index slightly to mimic how real human hands work
+            middle: input.rest_curl.max(index / 2.0),
+            ring: input.rest_curl.max(index / 4.0),
+            pinky: input.rest_curl.max(index / 6.0),
+            thumb: if input.thumb_touch { 1.0 } else { 0.0 },
+            time: current_time,
+        };
+
+        let elapsed_time = current_time.duration_since(state.time).as_secs_f32();
+        let alpha = (elapsed_time * 24.0).min(1.0);
```

Comment:
```suggestion
        let t = (elapsed_time * 24.0).min(1.0);
```
`alpha` elicits color terminology to me.